### PR TITLE
feat(sdk): move non-auth config options to provider

### LINKED ISSUE #50486: Move non-auth sdk config to MetabaseProvider props and rename field to authConfig
We should rename the current `SDKConfig` type to `MetabaseAuthConfig` and only store authentication-related config there. We should then move the rest of the props (errorComponent, loaderComponent, allowConsoleLog) to `MetabaseProvider`. We should call the configuration prop `authConfig`.

### PR DESCRIPTION
**Pull request by heypoom**
Closes https://github.com/metabase/metabase/issues/50486

- Renames `SDKConfig` to `MetabaseAuthConfig`
- Renames the `config` prop in MetabaseProvider to `authConfig`
- Moves the 3 fields not related to auth (i.e. `errorComponent`, `loaderComponent` and `allowConsoleLog`) out of the auth config, and into the MetabaseProvider's props.

### Checklist

- [x] Tests have been added/updated to cover changes in this PR

### CODE REVIEW

File: enterprise/frontend/src/embedding-sdk/CHANGELOG.md
Comment (WiNloSt) [inline enterprise/frontend/src/embedding-sdk/CHANGELOG.md:74]: Hmmm. This is the changelog, I don't think we should modify this. We should keep this for historical reason, and we'll have a new entry after this PR land that we change this to whatever the name we'll choose.

File: enterprise/frontend/src/embedding-sdk/components/public/MetabaseProvider.tsx
Comment (WiNloSt) [inline enterprise/frontend/src/embedding-sdk/components/public/MetabaseProvider.tsx:49]:
```suggestion
  /** Defines the display language. Accepts an ISO language code such as `en` or `de`. Defaults to `en`. */
```

File: enterprise/frontend/src/embedding-sdk/hooks/private/use-init-data.unit.spec.tsx
Comment (WiNloSt) [inline enterprise/frontend/src/embedding-sdk/hooks/private/use-init-data.unit.spec.tsx:42]: nit: I think we should also rename this

```suggestion
const TestComponent = ({ authConfig }: { authConfig: MetabaseAuthConfig }) => {
```

File: enterprise/frontend/src/embedding-sdk/lib/usage-problem.ts
Comment (WiNloSt) [inline enterprise/frontend/src/embedding-sdk/lib/usage-problem.ts:10]: I don't see this is used in this file?

File: enterprise/frontend/src/embedding-sdk/test/auth-flow.unit.spec.tsx
Comment (WiNloSt) [inline enterprise/frontend/src/embedding-sdk/test/auth-flow.unit.spec.tsx:45]: nit: 
```suggestion
    <MetabaseProvider authConfig={authConfig} locale={locale}>
```

File: enterprise/frontend/src/embedding-sdk/test/auth-flow.unit.spec.tsx
Comment (WiNloSt) [inline enterprise/frontend/src/embedding-sdk/test/auth-flow.unit.spec.tsx:89]:
```suggestion
      <MetabaseProvider authConfig={authConfig}>
```

File: enterprise/frontend/src/embedding-sdk/test/sdk-compatibility-warning.unit.spec.tsx
Comment (WiNloSt) [inline enterprise/frontend/src/embedding-sdk/test/sdk-compatibility-warning.unit.spec.tsx:26]: I'd suggest we rename this to:

```suggestion
const defaultAuthConfig = defineEmbeddingSdkAuthConfig({
```

or

```suggestion
const defaultSdkAuthConfig = defineEmbeddingSdkAuthConfig({
```

based on `DEFAULT_SDK_AUTH_PROVIDER_CONFIG` and `DEFAULT_AUTH_CONFIG` that we use in this PR.

File: enterprise/frontend/src/embedding-sdk/test/sdk-config-errors.unit.spec.tsx
Comment (WiNloSt) [inline enterprise/frontend/src/embedding-sdk/test/sdk-config-errors.unit.spec.tsx:30]: same as the previous comment, I think we should rename this to
```suggestion
const defaultAuthConfig = defineEmbeddingSdkAuthConfig({
```

File: enterprise/frontend/src/embedding-sdk/test/sdk-config-errors.unit.spec.tsx
Comment (WiNloSt) [inline enterprise/frontend/src/embedding-sdk/test/sdk-config-errors.unit.spec.tsx:116]:
```suggestion
      const authConfig = defineEmbeddingSdkAuthConfig({
```

**Review by WiNloSt — COMMENTED** [review]
Looks almost ready. I think one thing we should definitely fix is the changelog, other than that it's up to your discretion.

File: docs/embedding/sdk/config.md
Comment (heypoom) [reply docs/embedding/sdk/config.md:38]: @WiNloSt yes, I just noticed the props have been missing so I added it 😅

File: enterprise/frontend/src/embedding-sdk/CHANGELOG.md
Comment (heypoom) [reply enterprise/frontend/src/embedding-sdk/CHANGELOG.md:74]: Thanks for catching this! I must have forgotten to exclude this from find-and-replace.

**Comment by heypoom** [comment]
@WiNloSt Thanks for the careful review comments, it's always great to have a second pair of eyes when renaming things! I've resolved the issues in the following commits 🙏🏻

File: enterprise/frontend/src/embedding-sdk/hooks/private/use-init-data.unit.spec.tsx
Comment (oisincoveney) [inline enterprise/frontend/src/embedding-sdk/hooks/private/use-init-data.unit.spec.tsx:49]: could we use `createMockAuthProviderUriConfig` here too since we're already using it in the test scenarios?

**Review by oisincoveney — APPROVED** [review]
just make sure you get the approval from product + Jeff, but lgtm

File: docs/embedding/sdk/appearance.md
Comment (WiNloSt) [inline docs/embedding/sdk/appearance.md:16]:
```suggestion
import { defineMetabaseTheme } from "@metabase/embedding-sdk-react";
```

Small adjustment. I wonder if there's a formatter for code blocks like this 🤔 This isn't something we should have to be aware of.

File: enterprise/frontend/src/embedding-sdk/components/public/MetabaseProvider.tsx
Comment (WiNloSt) [inline enterprise/frontend/src/embedding-sdk/components/public/MetabaseProvider.tsx:49]: [there's a bug](https://metaboat.slack.com/archives/C063Q3F1HPF/p1733503500527059?thread_ts=1733501696.469049&cid=C063Q3F1HPF) currently that the SDK doesn't support locale like `en-US`.

We should include such a format to provide more clarity once we fix the bug.

**Review by WiNloSt — APPROVED** [review]
thanks for addressing the feedback ❤️

File: enterprise/frontend/src/embedding-sdk/components/public/MetabaseProvider.tsx
Comment (heypoom) [reply enterprise/frontend/src/embedding-sdk/components/public/MetabaseProvider.tsx:49]: @WiNloSt Good catch, let me note down that country code suffixes such as `en-US` is not supported.

Comment (heypoom) [reply enterprise/frontend/src/embedding-sdk/components/public/MetabaseProvider.tsx]: Clarified in https://github.com/metabase/metabase/pull/50585/commits/d35e5fc13122b010db91eabfefd74d8867ab8ad4

**Comment by heypoom** [comment]
@metabase-bot backport release-x.51.x

**Pull request merged**
